### PR TITLE
feat(representation): add segment slider and pull to refresh

### PR DESCRIPTION
**Pull request by thielpa**
add segment slider and pull to refresh

### CODE REVIEW

**Comment by Lordnoname** [comment]
@DevVersion Well in this pr went something wrong. I removed code and added the same one again?! Funny enough!

File: app/pages/representation/representation.ts [hunk 1]
```diff
@@ -61,6 +77,28 @@ export class RepresentationPage {
...
+          data: this.todayPromise,
+          key: 'today'
+        },
+        {
+          data: this.tomorrowPromise,
+          key: 'tomorrow'
+        },
+      ];
+    });
+    if (refresher !== undefined) refresher.complete();
```
Comment (devversion) [inline app/pages/representation/representation.ts]: Can be simplified to `if (refresher)`

File: app/pages/representation/representation.ts [hunk 2]
```diff
@@ -61,6 +77,28 @@ export class RepresentationPage {
...
+  refreshRepresentations(session: SessionAccessor, backend: BackendConnector, refresher?) {
+    this.isLoadingToday = true;
+    this.isLoadingTomorrow = true;
+
+    session.getToken().then((token) => {
+      (this.todayPromise = backend.sendRepresentationRequest(this.todayDate, token)).then(success => this.isLoadingToday = false, error => this.isLoadingToday = false);
+      (this.tomorrowPromise = backend.sendRepresentationRequest(this.tomorrowDate, token)).then(success => this.isLoadingTomorrow = false, error => this.isLoadingTomorrow = false);
+
+      this.slides = [
+        {
```
Comment (devversion) [inline app/pages/representation/representation.ts]: Braces should start the following

``` ts
this.slides = [{

}, {

}];
```

File: app/pages/representation/representation.ts [hunk 3]
```diff
@@ -61,6 +77,28 @@ export class RepresentationPage {
     });
   }
 
+  refreshRepresentations(session: SessionAccessor, backend: BackendConnector, refresher?) {
+    this.isLoadingToday = true;
+    this.isLoadingTomorrow = true;
+
+    session.getToken().then((token) => {
+      (this.todayPromise = backend.sendRepresentationRequest(this.todayDate, token)).then(success => this.isLoadingToday = false, error => this.isLoadingToday = false);
+      (this.tomorrowPromise = backend.sendRepresentationRequest(this.tomorrowDate, token)).then(success => this.isLoadingTomorrow = false, error => this.isLoadingTomorrow = false);
```
Comment (devversion) [inline app/pages/representation/representation.ts]: Wrap at 100 columns

File: app/pages/representation/representation.ts [hunk 4]
```diff
@@ -61,6 +77,28 @@ export class RepresentationPage {
     });
   }
 
+  refreshRepresentations(session: SessionAccessor, backend: BackendConnector, refresher?) {
+    this.isLoadingToday = true;
+    this.isLoadingTomorrow = true;
```
Comment (devversion) [inline app/pages/representation/representation.ts]: IMO the refresher should only refresh the tab which was pulled out.

**Comment by thielpa** [comment]
@DevVersion pr updated

**Comment by devversion** [comment]
@Lordnoname Is pull to refresh still bugging?

**Comment by thielpa** [comment]
This is an issue from ionic 2. I can't fix that

**Comment by devversion** [comment]
Okay, but then we should remove that, if it's still bugging.

**Comment by thielpa** [comment]
@DevVersion I removed it.^
BTW: its not the pull to refresh thats bugging, but the ion-spinner

**Comment by thielpa** [comment]
@DevVersion resolved conflicts. Ready for merge?

**Comment by Lordnoname** [comment]
@DevVersion resolved conflicts

**Comment by thielpa** [comment]
@DevVersion Depends on #25 (angular-beta.4)

**Comment by pgschwendtner** [comment]
Thanks for your reply, we will analyze your comment as soon as possible.

File: app/pages/representation/representation.scss [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+}
+
+[menuLeft] {
+  min-height: 100%;
+  padding-left: 5px;
+}
+
+[hidden] {
+  display: none;
+}
```
Comment (devversion) [inline app/pages/representation/representation.scss:15]: Attribute Selectors are slow IE11

Comment (thielpa) [inline app/pages/representation/representation.scss:15]: Well.....good joke :P

**Pull request merged**